app/main.py

```diff
@@ -12,7 +45,125 @@
     allow_headers=["*"],
 )
 
+class Request(BaseModel):
+    location: str
+    prompt:  str
+
+class Response(BaseModel):
+    id: int
+    name: str
+    long: float
+    lat: float
+    area: str
+    category: str
 
 @app.get("/")
 async def hello():
-    return {"message": "Hello World"}
\ No newline at end of file
+    return {"message": "Hello World"}
+
+@app.post("/api")
+async def search_pub(request: Request):
+    csvlist_by_python = []
+    csvlist_by_chroma = []
+    distancelist = []
+
+    location = request.location
+    prompt = request.prompt
+
+    lat, lng = await get_cordinates(location)
+
+    if lat and lng is None:
+        return {"error": "Invalid location"}
+    try:
+        with open(csv_filepath, encoding="utf-8", newline="") as f:
+            reader=csv.reader(f)
+            for index, row in enumerate(reader):
+                csvlist_by_python.append(row)
+                # 1行目はヘッダーなのでスキップ
+                if  index == 0:
+                    continue
+
+                csvlist_by_python[index][LONG_INDEX], csvlist_by_python[index][LAT_INDEX] = float(csvlist_by_python[index][LONG_INDEX]), float(csvlist_by_python[index][LAT_INDEX])
+
+                # distancelistにpopしたいindexをメモ
+                distance = math.sqrt((lat - csvlist_by_python[index][LONG_INDEX])**2 + (lng - csvlist_by_python[index][LAT_INDEX])**2)
+
+                if distance > 0.006:
+                    distancelist.append(index)
```

Comment:
`0.006`このようなコード内で使用される具体的な数値をマジックナンバーと言います。マジックナンバーとはその値が具体的に何を表す数値なのかわからない数のことです。今回だと自分が事前に数値のスケールを聞いていたので`0.006`が何を表していてどれくらいの大きさのことを言うのか知っているのでこの評価を理解できますが、極力変数に入れたり値の単位がわかるようにしておくとわかりやすいですね！緯度経度を用いた距離計算の単位は度(`degrees`)なので`distance`ではなく`distance_degrees`などにしておくとわかりやすいです。これで「度」の単位で比較していることがわかるためマジックナンバーのスケールが具体的になります。数値比較の場合は変数名に単位を入れることがリーダブルコードやクリーンコードでは推奨されてるので気になるようであれば調べてみてください！

```suggestion
                if distance_degrees > 0.006:
                    distancelist.append(index)
```

Reply:
ちなみに「度」の計算だとスケールがわかりずらいという場合はキロメートルに変換する方法もありますね！。緯度は1度あたり112km弱、経度は緯度にcosをとれば出すことができますね！幅が狭いので今回はあんまり恩恵を受けれないのであまり使う必要性はないが。。。

Reply:
これたしかにめっちゃ分かりづらい🤔
座標で2点間の距離を評価するの意味分からん気がしてきたからメートル単位に変換しちゃおうと思う💪
geographiclibライブラリを使って関数作ってみた！
```
def calculate_distance(lat1, lon1, lat2, lon2):
    geod = Geodesic.WGS84
    g = geod.Inverse(lat1, lon1, lat2, lon2)
    return g['s12']
```
この関数の返り値はメートル単位のfloat型の数値で返るようになってる！

Reply:
`g['12']`が返り値っていうのはわかりずらいので一旦`distance_meters = g['12']`に格納してから`distance_meters`をreturnする方が綺麗じゃないですかね。イメージとしては以下です：
```py
def calculate_distance(current: Coordinate, destination: Coordinate) -> float:
    geod = Geodesic.WGS84
    g = geod.Inverse(*current, *destination)
    distance_meters = g['12']
    return distance_meters
```

他のレビューコメントでも指摘しましたが、座標注入はタプル型で行いましょう。
`g = geod.Inverse(*current, *destination)`この部分でタプルをそれぞれ緯度経度に展開しています。

Reply:
質問！！！
currentとdestinationはクラスで定義してるから、`*`でアンパックすると,
```
('lat', 35.681236) ('lng', 139.767125)
```
(数値はダミー)
こんな感じになっちゃう。。。
座標注入をタプル型でやるなら、currentとdestinationをCoordinate型のクラスじゃなくてタプル型で定義してやる方がいいんじゃないだろうか🤔
アンパックの仕方ミスってたりするんだったらすまん！

Reply:
最終的にCoodinateクラスを下に定義した！
```
class Coordinate(BaseModel):
    coordinate: tuple[float, float]
```
Coodinateクラスの属性の型をタプルにすることで実現できた！
```
def calculate_destination_distance(search_point_coodinate: Coordinate, izakaya_coordinate: Coordinate) -> float:
    geod = Geodesic.WGS84
    g = geod.Inverse(*search_point_coodinate.coordinate, *izakaya_coordinate.coordinate)
    destination_distance_meters = g['s12']
    return destination_distance_meters
```